charts/nginx-gateway-fabric/values.yaml

```diff
@@ -104,12 +104,18 @@ terminationGracePeriodSeconds: 30
 ## Tolerations for the NGINX Gateway Fabric pod.
 tolerations: []
 
-##  The nodeSelector of the NGINX Gateway Fabric pod.
+## The nodeSelector of the NGINX Gateway Fabric pod.
 nodeSelector: {}
 
 ## The affinity of the NGINX Gateway Fabric pod.
 affinity: {}
 
+## The platform that NGINX Gateway Fabric is being deployed on. Valid values: kubernetes, openshift
+platform: kubernetes
+
+## The name for the SecurityContextConstraints object used in Openshift environments.
+sccName: nginx-ngf-admin
```

Comment:
Could we omit this and just follow a similar pattern that we use to generate the other CRD names that we use? https://github.com/nginxinc/nginx-gateway-fabric/blob/main/charts/nginx-gateway-fabric/templates/_helpers.tpl#L26

Reply:
Is there a reason why we don't list the other CRD names in the values file and instead require them to be kind of hidden (nameOverride isn't documented anywhere)? Is it because we don't expect the CRD names to need to be changed much? 

If so, then if the SCC falls into the same thought of not needing to be changed often, I think we could do that. But some of the suggestions above make me err on maybe they would need a custom name for the SCC more often.

Reply:
The generated names use the release name of the helm installation as part of the name, which will help avoid name collisions. It likely won't be a common case where a user needs to change it.

Reply:
ok sure lemme make the change and we'll see how it looks

Reply:
Made the changes, here is the output of not setting the nameOverride like such:

`helm template ngf ./charts/nginx-gateway-fabric --create-namespace -n nginx-gateway --set nginxGateway.productTelemetry.enable=false --set platform=openshift`

```
rbac.yaml: 

- apiGroups:
  - security.openshift.io
  resources:
  - securitycontextconstraints
  resourceNames:
  - ngf-scc
  verbs:
  - use
  
  ...
  
  scc.yaml
  
  # Source: nginx-gateway-fabric/templates/scc.yaml
# Create SCC for Gateway resources
kind: SecurityContextConstraints
apiVersion: security.openshift.io/v1
metadata:
  name: ngf-scc
allowPrivilegeEscalation: false
allowHostDirVolumePlugin: false
....
```

here is the output of setting nameOverride like such:

`helm template ngf ./charts/nginx-gateway-fabric --create-namespace -n nginx-gateway --set nginxGateway.productTelemetry.enable=false --set platform=openshift --set nameOverride=this-is-new-name`

```
rbac.yaml:

- apiGroups:
  - security.openshift.io
  resources:
  - securitycontextconstraints
  resourceNames:
  - this-is-new-name-scc
  verbs:
  - use

scc.yaml:

# Source: nginx-gateway-fabric/templates/scc.yaml
# Create SCC for Gateway resources
kind: SecurityContextConstraints
apiVersion: security.openshift.io/v1
metadata:
  name: this-is-new-name-scc
allowPrivilegeEscalation: false
allowHostDirVolumePlugin: false

...
```